lib/magic-string.js

```diff
@@ -0,0 +1,142 @@
+"use strict";
+
+// A simplified version of magic-string.
+// Copyright Rich Harris. Released under MIT license:
+// https://github.com/Rich-Harris/magic-string
+
+class Chunk {
+  constructor(start, end, content) {
+    this.start = start;
+    this.end = end;
+    this.original = content;
+    this.intro = "";
+    this.outro = "";
+    this.content = content;
+    this.next = null;
+  }
+
+  appendLeft(content) {
```

Comment:
No space after a function or method name. I only put a space before the `(` if it's an anonymous function.